packages/plugins/typescript/urql/tests/urql.spec.ts

```diff
@@ -590,6 +590,147 @@ export function useSubmitRepositoryMutation() {
+        {
+          outputFile: 'graphql.tsx',
+        },
+      ));
+
+      expect(content.content).toBeSimilarStringTo(`
+      export function useFeedQuery(options: Omit<Urql.UseQueryArgs<IFeedQueryVariables>, 'query'> = {}) {
+        const context = useMemo(() => ({
+          ...(options.context || {}),
+          additionalTypenames: [Entry],
+        }, [options.context]));
+        return Urql.useQuery<IFeedQuery>({ query: FeedDocument, ...options, context });
+      };
+      `);
+    });
+
+    it('should correctly generate nested additionalTypenames', async () => {
+      const documents = parse(/* GraphQL */ `
+        query feed {
+          feed {
+            id
+            repository {
+              full_name
+              owner {
+                avatar_url
+              }
+            }
+          }
+        }
+      `);
+
+      const docs = [{ location: '', document: documents }];
+
+      const content = (await plugin(
+        schema,
+        docs,
+        { withHooks: true, typesPrefix: 'I', withAdditionalTypenames: true },
+        {
+          outputFile: 'graphql.tsx',
+        },
+      ));
+
+      expect(content.content).toBeSimilarStringTo(`
+      export function useFeedQuery(options: Omit<Urql.UseQueryArgs<IFeedQueryVariables>, 'query'> = {}) {
+        const context = useMemo(() => ({
+          ...(options.context || {}),
+          additionalTypenames: [Entry, Repository, User],
+        }, [options.context]));
+        return Urql.useQuery<IFeedQuery>({ query: FeedDocument, ...options, context });
+      };
+      `);
+    });
+
+    it('should correctly generate additionalTypenames with fragments', async () => {
+      const documents = parse(/* GraphQL */ `
+        query feed {
+          feed {
+            ...MyEntry
+          }
+        }
+
+        fragment MyEntry on Entry {
+          id
+        }
+      `);
+
+      const docs = [{ location: '', document: documents }];
+
+      const content = (await plugin(
+        schema,
+        docs,
+        { withHooks: true, typesPrefix: 'I', withAdditionalTypenames: true },
+        {
+          outputFile: 'graphql.tsx',
+        },
+      ));
+
+      expect(content.content).toBeSimilarStringTo(`
+      export function useFeedQuery(options: Omit<Urql.UseQueryArgs<IFeedQueryVariables>, 'query'> = {}) {
+        const context = useMemo(() => ({
+          ...(options.context || {}),
+          additionalTypenames: [Entry],
+        }, [options.context]));
+        return Urql.useQuery<IFeedQuery>({ query: FeedDocument, ...options, context });
+      };
+      `);
+    });
+
+    it('should correctly generate additionalTypenames with fragments (with nested types)', async () => {
+      const documents = parse(/* GraphQL */ `
+        query feed {
+          feed {
+            ...MyEntry
+          }
+        }
+
+        fragment MyEntry on Entry {
+          id
+          repository {
```

Comment:
Do we support doing this

Reply:
@dotansimha it seems like we aren't sending the `fragmentDefinitions` along to the `_buildX` methods. Is this possible? If so do we expect that to happen frequently?

Reply:
I think it's something you can add :)